test/compile-test.cc

```diff
@@ -316,6 +316,19 @@ TEST(compile_test, compile_format_string_literal) {
 }
 #endif
 
+#if defined(__cpp_if_constexpr) && defined(__cpp_return_type_deduction)
+template <typename S>
+typename fmt::is_compiled_string<S>::value_type check_is_compiled_string(
+    const S&) {
```

Comment:
`value_type` is just `bool` so let's just use that for simplicity.

Reply:
Done